frontends/concrete-python/examples/floating_point/floating_point.ipynb

```diff
@@ -0,0 +1,609 @@
+   "metadata": {},
+   "source": [
+    "We can remark that here, the function is much slower, since it includes PBS. Furthermore, the input of the PBS is as large as twice the square of typical inputs, which can quickly become large. That's why we have had to reduce the scaling factor and the precision. When more precision is needed, one would have to be a bit more patient."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "85995459-8036-4b18-92f0-261d524b1a6e",
+   "metadata": {},
+   "source": [
+    "## Final example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "afc5c7eb-db95-49c7-accc-5897e53db414",
+   "metadata": {},
+   "source": [
+    "We finish with a more complex example, where one needs to understand what kind of computations are done and cancel some scaling factors during the computation, to ensure correct computations.\n",
+    "\n",
+    "Let's suppose we want to convert `f(x, y) = (x**2 + 4y) / 1.33` to FHE. What we'll do is that, we'll scale `x` and `y` with a scaling factor `scaling_factor`, before encryption. Then, in FHE, `x**2` will somehow have `scaling_factor**2` as a scaler while `4y` will only have `scaling_factor`: so, to keep the addition homogeneous, we need to multiply `4y` by `scaling_factor` in the FHE circuit. Remark we could also have divided `x**2` by `scaling_factor` to keep things homogeneous. \n",
+    "\n",
+    "We continue like this, to know the scaler of all the values in the circuit, and keep things homogeneous. At the end of the FHE function, we see that the scaler is `scaling_factor**2`, so we need to divide by `scaling_factor**2` after the decryption.\n",
+    "\n",
+    "The error bound is more complicated to compute: we can use an estimation.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "e3fdb20c-3e28-489e-8ede-640331674b7a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Maximal bitwidth in the circuit: 10\n",
+      "\n",
+      "Showing an example\n",
+      "random_sample=(1.1831267112584711, 0.18990444363541092)\n",
+      "scaled_sample=(9, 2)\n",
+      "scaled_result=109\n",
+      "\n",
+      "FHE:   1.703125\n",
+      "clear: 1.6236139770187439\n",
+      "bounded error: 0.375\n",
+      "\n",
+      "Compilation and test look good, with FHE execution time of about 36.05 seconds per inference\n"
+     ]
+    }
+   ],
+   "source": [
+    "max_value_for_floats = 2\n",
+    "scaling_factor = 8\n",
+    "nb_test_samples = 8\n",
+    "\n",
+    "def special_function_in_clear(x, y):\n",
+    "    u = x**2\n",
+    "    v = u + 4 * y\n",
+    "    return v / 1.33\n",
+    "\n",
+    "@fhe.compiler({\"x\": \"encrypted\", \"y\": \"encrypted\"})\n",
+    "def special_function(x, y):\n",
+    "    u = x**2\n",
+    "    v = u + 4 * y * scaling_factor\n",
+    "    return np.round(fhe.univariate(lambda x: x / 1.33)(v)).astype(np.int64)\n",
```

Comment:
Instead of scaling `v` to `scaling_factor**2`, why not reduce the scaling when applying `x**2` tlu?
```python
u = fhe.univariate(lambda x: (x**2) // scaling_factor)(x)
```

Reply:
yes, it's doable, I have already said "Remark we could also have divided `x**2` by `scaling_factor` to keep things homogeneous."

What I liked with keeping it like I mentioned was: showing that, at the end, we have to divide by scaling_factor**2 this time, it's not always scaling_factor. If we were to do as you say, people may think it's always scaling_factor.

So, ok to keep as I say, @umut-sahin ?

Reply:
I think the benefit is that doing it this way would result in reduced bit-width. So this should be what we show IMO. Can you turn it around and mention the other approach in text and use this in code?

Reply:
ok let me do it

Reply:
done, ok with you?